### ISSUE TITLE
Come up with a good way to support backends w/o poluting deps

### ISSUE THREAD

**Issue by oubiwann** [lfex/lhc#32]
I'm thinking default = Erlang httpd. Simply calling `(lhc:start)` would use that. If you wanted to use a different backend, you could pas the app name in `start/1`: `(lhc:start #m(backend hackney))`. It would be up to the user to add the `lhc-hackney` dep in their `rebar.config` (if they hadn't, start should fail).

**Comment by oubiwann** [comment]
In fact, the next release can go back to just Erlang httpc support, and subsequent releases can add support for the different client backends/plugins.

**Comment by John-Goff** [comment]
This is very similar to the function of the quite well designed [Tesla](https://hexdocs.pm/tesla/readme.html) library in Elixir. The way they handle it is by having a [behaviour](https://hexdocs.pm/tesla/Tesla.Adapter.html) that needs to be implemented for any backend you want to include which has a single `call/1` function. They have a single [struct](https://hexdocs.pm/tesla/Tesla.Env.html) which represents the request and the response, but the `lhc` version could accept a `http.request` map and return a `http.response`. Then the behaviour implementation can either be included in a separate library, or [the way Tesla does it](https://github.com/elixir-tesla/tesla/blob/d3ea190250f2807077dcc32d0d6ff85ea27b671a/lib/tesla/adapter/finch.ex#L1) is to wrap the entire module definition in a call to `Code.ensure_loaded?/1`, which is similar to what I did [here](https://github.com/lfe/rebar3/blob/672806f61e82bec548e30ff423b8ceb96fdc3647/src/rebar3_lfe_prv_ltest.erl#L37) for the rebar3_lfe plugin and ltest.

**Comment by oubiwann** [comment]
Yeah, I just ran across something similar with the fs (file watcher) Erlang lib -- this was done for compiled binaries (which differed in name and capability for different architectures). But they same thing (behaviour, impls, etc.).

Love it!
